db/seeds.rb

```diff
@@ -68,161 +68,193 @@ def fact_check_attributes(fact_check_link, user, project, team)
+
+    puts 'Making Project...'
+    project = create_project(title: team.name, team_id: team.id, user: user, description: '')
+
+    puts 'Making Team User...'
+    create_team_user(team: team, user: user, role: 'admin')
+  end
+
+  if answer == "2"
+    puts "Type user name then press enter"
+    print ">> "
+    name = STDIN.gets.chomp.downcase
+
+    puts "Fetching User, Project, Team User and Team..."
+    user = User.find_by(name: name)
+    project = Project.find_by(user_id: user.id)
+    team_user = TeamUser.find_by(user_id: user.id)
+    team = Team.find_by(id: team_user.team_id)
+  end
+
+  puts 'Making Medias...'
+  puts 'Making Medias and Project Medias: Claims...'
+  9.times { Claim.create!(user_id: user.id, quote: Faker::Quotes::Shakespeare.hamlet_quote) }
+  create_project_medias(user, project, team)
+  add_claim_descriptions_and_fact_checks(user)
+
+  puts 'Making Medias and Project Medias: Links...'
+  begin
+    data[:link_media_links].each { |link_media_link| Link.create!(user_id: user.id, url: link_media_link+"?timestamp=#{Time.now.to_f}") }
+    create_project_medias(user, project, team)
+    add_claim_descriptions_and_fact_checks(user)
+  rescue
+    puts "Couldn't create Links. Other medias will still be created. \nIn order to create Links make sure Pender is running."
+  end
+
+  puts 'Making Medias and Project Medias: Audios...'
+  data[:audios].each { |audio| UploadedAudio.create!(user_id: user.id, file: open_file(audio)) }
+  create_project_medias(user, project, team)
+  add_claim_descriptions_and_fact_checks(user)
+
+  puts 'Making Medias and Project Medias: Images...'
+  data[:images].each { |image| UploadedImage.create!(user_id: user.id, file: open_file(image))}
+  create_project_medias(user, project, team)
+  add_claim_descriptions_and_fact_checks(user)
+
+  puts 'Making Medias and Project Medias: Videos...'
+  data[:videos].each { |video| UploadedVideo.create!(user_id: user.id, file: open_file(video)) }
+  create_project_medias(user, project, team)
+  add_claim_descriptions_and_fact_checks(user)
+
+  puts 'Making Claim Descriptions and Fact Checks: Imported Fact Checks...'
+    data[:fact_check_links].each { |fact_check_link| create_fact_check(fact_check_attributes(fact_check_link, user, project, team)) }
+
+  if answer == "1"
+    puts 'Making Relationship between Claims...'
+    relationship_claims = []
+    project_medias_for_relationship_claims = []
+    data[:quotes].each { |quote| relationship_claims.push(Claim.create!(user_id: user.id, quote: quote)) }
+    relationship_claims.each { |claim| project_medias_for_relationship_claims.push(ProjectMedia.create!(user_id: user.id, project: project, team: team, media: claim))}
+
+    Relationship.create!(source_id: project_medias_for_relationship_claims[0].id, target_id: project_medias_for_relationship_claims[1].id, relationship_type: Relationship.confirmed_type)
```

Comment:
Good question, makes me think: what is critical and what could we let fail?

Right now we are creating all types of medias, imported fact checks, relationships and tip line requests. Which ones are critical and which ones aren't? The only one we let fail are Links, but mostly because of their Pender dependency.

Reply:
I think the transaction + create! works well for now, so let's leave as-is, but definitely good to think in future if we end up making this more complex what is necessary vs nice-to-have